test/unit/CMakeLists.txt

```diff
@@ -85,3 +85,20 @@ target_include_directories(boost_url_tests PRIVATE . ../../extra/include)
 target_link_libraries(boost_url_tests PRIVATE Boost::url)
 add_test(NAME boost_url_tests COMMAND boost_url_tests)
 add_dependencies(tests boost_url_tests)
+
+if (CMAKE_CXX_STANDARD GREATER_EQUAL 14)
+    include(FetchContent)
+    FetchContent_Declare(
+            fmt
+            GIT_REPOSITORY https://github.com/fmtlib/fmt
+            GIT_TAG        8.1.1
+    )
+    FetchContent_MakeAvailable(fmt)
+    target_link_libraries(boost_url_tests PRIVATE fmt::fmt)
+    if (CMAKE_CXX_STANDARD GREATER_EQUAL 17)
```

Comment:
The logic here, for now, was to recur to `__has_include` in C++17 and to a `BOOST_URL_HAS_FMT_FORMAT` macro otherwise.